dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/importers/ui/ImporterForm.java

```diff
@@ -0,0 +1,254 @@
+                    ? FormConstants.MULTIPLE_OPTIONS
+                    : FormConstants.SINGLE_OPTIONS;
+     }
+
+    // -------------------- LOGIC --------------------
+
+    public static ImporterForm createInitializedForm(MetadataImporter importer, Locale locale,
+                                                     Supplier<Map<MetadataBlock, List<DatasetFieldsByType>>> metadataSupplier) {
+        ImporterForm instance = new ImporterForm();
+        instance.initializeForm(importer, locale,
+                MetadataFormLookup.create(importer.getMetadataBlockName(), metadataSupplier));
+        return instance;
+    }
+
+    public void initializeForm(MetadataImporter importer, Locale locale, MetadataFormLookup lookup) {
+        this.lookup = lookup;
+        this.importer = importer;
+        this.bundleWrapper = new SafeBundleWrapper(importer, locale);
+        initializeFormItems();
+    }
+
+    public void handleFileUpload(FileUploadEvent event) throws IOException {
+        FileUpload component = (FileUpload) event.getComponent();
+        removePreviousTempFile(component);
+
+        UploadedFile file = Optional.ofNullable(event)
+                .map(FileUploadEvent::getFile)
+                .orElseThrow(() -> new IllegalStateException("Null event or file"));
+        Path tempPath = prepareTempPath(file);
+        Files.copy(file.getInputStream(), tempPath, StandardCopyOption.REPLACE_EXISTING);
+        component.setValue(tempPath.toFile());
+    }
+
+    public void nextStep() {
+        Set<ValidationResult> validationResults = validateFormInputs();
+        if (!validationResults.isEmpty()) {
+            showValidationMessages(validationResults);
+            return;
+        }
+        List<ResultField> resultFields = importer.fetchMetadata(toImporterInput(items));
+        resultItems = new ResultItemsCreator(lookup).createItemsForView(resultFields);
+        step = ImportStep.SECOND;
+    }
+
+    public void onExit(Map<MetadataBlock, List<DatasetFieldsByType>> metadata) {
+        cleanUp();
+        new MetadataFormFiller(lookup).fillForm(resultItems);
+    }
+
+    public void cleanUp() {
+        items.stream()
+                .filter(i -> i.getType().equals(ImporterFieldType.UPLOAD_TEMP_FILE) && i.getValue() != null)
+                .forEach(i -> ((File) i.getValue()).delete());
+    }
+
+    // -------------------- PRIVATE --------------------
+
+    private void initializeFormItems() {
+        int counter = 1;
+        for (ImporterData.ImporterField field : getImporterFields(importer)) {
+            FormItem formItem = new FormItem(generateViewId(field, counter), field, bundleWrapper);
+            items.add(formItem);
+            keyToItem.put(field.fieldKey, formItem);
+            counter++;
+        }
+    }
+
+    private String generateViewId(ImporterData.ImporterField field, int ordinal) {
+        return String.join("_",
+                String.valueOf(Math.abs(field.fieldKey.hashCode() % 512)),
```

Comment:
1. Because I use the method that searches for the component by part of its id, I wanted to add something to that id, to be sure that (because I cannot control the names of the importer components that are also part of the id) the method would not accidentally return some other component because it would have the same name. 
2. Adding only an ordinal number seemed to weak, and I also wanted to add something that is not totally random.
3. So the best candidate was a hashcode, but I didn't wanted to present that unchanged in the view, so I take the absolute value (because some are negative) and – to limit the range – make a modulo division by 512.